ledger/catchupaccessor.go

```diff
@@ -128,6 +128,30 @@ func (w *stagingWriterImpl) writeKVs(ctx context.Context, kvrs []encoded.KVRecor
 		hashes := make([][]byte, len(kvrs))
 		for i := 0; i < len(kvrs); i++ {
 			keys[i] = kvrs[i].Key
+
+			// Since `encoded.KVRecordV6` is `omitempty` and `omitemptyarray`,
+			// when we have an instance of `encoded.KVRecordV6` with nil value,
+			// an empty box is unmarshalled to have `nil` value,
+			// while this might be mistaken to be a box deletion.
+			//
+			// We don't want to mistake this to be a deleted box:
+			// We are (and should be) during Fast Catchup (FC)
+			// writing to DB with empty byte string, rather than writing nil.
+			//
+			// This does not really matter in sqlite3,
+			// for sqlite3 treats them both as empty BLOB,
```

Comment:
I don't think this part of the explanation is right. We _did_ have a bug because we read them as `nil` (which they are). But @algorandskiy fixed this on the read path by converting `nil` to `[]byte{}`.

Reply:
yes, you are right, the sqlite3 does treat writing `nil` and `[]byte{}` as different things: see [update comment explaining](https://github.com/algorand/go-algorand/pull/5225/commits/7e696457ae3da859623f64d0e5aa1f247edb221e)